### PR TITLE
[DEV-1857][chore] init: add register_tutorial_api_token SDK command

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Add `register_tutorial_api_token` SDK command.

We should be able to place this at the top of the notebook

```
import featurebyte as fb
fb.register_tutorial_api_token("<api_token>")
```
and this will store the token in the config file, and change to the `tutorial` profile.

## Related Issue
https://featurebyte.atlassian.net/browse/DEV-1857

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/__init__.py [hunk 1]
```diff
@@ -165,6 +166,60 @@ def use_profile(profile: str) -> None:
...
+    updated_profile = False
+    # Update API token in existing profile if it's there
+    for profile in profiles:
+        if profile["name"] == tutorial_profile_name:
+            if "api_token" in profile:
+                updated_profile = profile["api_token"] != api_token
+            else:
+                updated_profile = True
+            profile["api_token"] = api_token
+            has_tutorial_profile = True
```
Comment (YS-L) [inline featurebyte/__init__.py:200]: nit: can consider breaking from this loop once tutorial profile name is found and replace `if not has_tutorial_profile:` with `else:` then there's no need for the variable `has_tutorial_profile`.

**Review by YS-L — APPROVED** [review]

File: featurebyte/__init__.py [hunk 2]
```diff
@@ -165,6 +166,60 @@ def use_profile(profile: str) -> None:
...
+    has_tutorial_profile = False
+    updated_profile = False
+    # Update API token in existing profile if it's there
+    for profile in profiles:
+        if profile["name"] == tutorial_profile_name:
+            if "api_token" in profile:
+                updated_profile = profile["api_token"] != api_token
+            else:
+                updated_profile = True
+            profile["api_token"] = api_token
```
Comment (YS-L) [inline featurebyte/__init__.py:199]: Btw do we have to update `api_url` to ensure that it's pointing to the correct endpoint?

File: featurebyte/__init__.py [hunk 3]
```diff
@@ -165,6 +166,60 @@ def use_profile(profile: str) -> None:
...
+            }
+        )
+        loaded_config["profile"] = profiles
+        updated_profile = True
+
+    # Write to config file if profile was updated
+    if updated_profile:
+        yaml_str = yaml.dump(loaded_config, sort_keys=False)
+        with open(config_file_path, "w", encoding="utf-8") as file_obj:
+            file_obj.write(yaml_str)
```
Comment (gee-senbong) [inline featurebyte/__init__.py:217]: Should we rename an existing profile into something like `confic.yaml.bak` rather than overwrite an existing profile?

**Review by gee-senbong — APPROVED** [review]
LGTM, left a minor comment to consider.

File: featurebyte/__init__.py (hunk 2 above)
Comment (jevonyeoh) [reply featurebyte/__init__.py:199]: hmm i guess i've assumed that the `api_url` is already pointing to `tutorials`, but no harm i guess!

File: featurebyte/__init__.py (hunk 3 above)
Comment (jevonyeoh) [reply featurebyte/__init__.py:217]: sounds good, i can do that.

File: featurebyte/__init__.py (hunk 1 above)
Comment (jevonyeoh) [reply featurebyte/__init__.py:200]: 😱 TIL. for-loop-break-else

**Pull request merged**